e2e/e2e_test.go

```diff
@@ -1525,3 +1536,298 @@ func TestPullRequestReviewDeletion(t *testing.T) {
+	require.Error(t, err)
+	require.Nil(t, sub)
+	require.Equal(t, 404, resp2.StatusCode)
+}
+
+func TestE2E_DismissNotification(t *testing.T) {
+	skipIfGlobalMutationNotOptedIn(t)
+	t.Parallel()
+	client := setupMCPClient(t)
+	ctx := context.Background()
+
+	// List notifications to get a valid threadID
+	listReq := mcp.CallToolRequest{}
+	listReq.Params.Name = "list_notifications"
+	listReq.Params.Arguments = map[string]any{}
+	resp, err := client.CallTool(ctx, listReq)
+	require.NoError(t, err)
+	require.False(t, resp.IsError)
+	textContent, ok := resp.Content[0].(mcp.TextContent)
+	require.True(t, ok)
+	var notifications []struct {
+		ID string `json:"id"`
+	}
+	err = json.Unmarshal([]byte(textContent.Text), &notifications)
+	require.NoError(t, err)
+	if len(notifications) == 0 {
+		t.Skip("No notifications available to test dismissal")
+	}
+	require.NotEmpty(t, notifications)
+	threadID := notifications[0].ID
+
+	// Dismiss notification (mark as read)
+	dismissReq := mcp.CallToolRequest{}
+	dismissReq.Params.Name = "dismiss_notification"
+	dismissReq.Params.Arguments = map[string]any{
+		"threadID": threadID,
+		"state":    "read",
+	}
+	resp, err = client.CallTool(ctx, dismissReq)
+	require.NoError(t, err)
+	require.False(t, resp.IsError)
+	textContent, ok = resp.Content[0].(mcp.TextContent)
+	require.True(t, ok)
+	require.Contains(t, textContent.Text, "read")
+}
+
+func TestE2E_MarkAllNotificationsRead(t *testing.T) {
+	skipIfGlobalMutationNotOptedIn(t)
+	t.Parallel()
+	client := setupMCPClient(t)
+	ctx := context.Background()
+
+	// Limit to notifications updated within the last hour
+	oneHourAgo := nowMinusOneHourRFC3339()
+	markAllReq := mcp.CallToolRequest{}
+	markAllReq.Params.Name = "mark_all_notifications_read"
+	markAllReq.Params.Arguments = map[string]any{
+		"since": oneHourAgo,
+	}
+	resp, err := client.CallTool(ctx, markAllReq)
+	require.NoError(t, err)
+	require.False(t, resp.IsError)
+	textContent, ok := resp.Content[0].(mcp.TextContent)
+	require.True(t, ok)
+	require.Contains(t, textContent.Text, "All notifications marked as read")
+
+}
+
+// nowMinusOneHourRFC3339 returns the RFC3339 timestamp for one hour ago from now (UTC)
+func nowMinusOneHourRFC3339() string {
+	return time.Now().UTC().Add(-1 * time.Hour).Format(time.RFC3339)
+}
+
+func TestE2E_GetNotificationDetails(t *testing.T) {
+	t.Parallel()
+	client := setupMCPClient(t)
+	ctx := context.Background()
+
+	// List notifications to get a valid notificationID
+	listReq := mcp.CallToolRequest{}
+	listReq.Params.Name = "list_notifications"
+	listReq.Params.Arguments = map[string]any{}
+	resp, err := client.CallTool(ctx, listReq)
+	require.NoError(t, err)
+	require.False(t, resp.IsError)
+	textContent, ok := resp.Content[0].(mcp.TextContent)
+	require.True(t, ok)
+	var notifications []struct {
+		ID string `json:"id"`
+	}
+	err = json.Unmarshal([]byte(textContent.Text), &notifications)
+	require.NoError(t, err)
+	require.NotEmpty(t, notifications)
```

Comment:
I think this probably needs:

```
	if len(notifications) == 0 {
		t.Skip("No notifications available to test dismissal")
	}
```

The test failed for me in a surprising way because I had no notifications.